From 030b8666f0ed5a4aa9c56487b9b3cebfa83328e6 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 28 Apr 2022 13:44:12 +0300
Subject: [PATCH] lx2160a_common: change ramdisk_addr_r to after kernel_addr_r

To allow for larger ramdisks, change the load address from
0x09000000 to 0x85100000 - which is exactly 64MB after kernel_addr_r.

This solves errors when loading a large ramdisk, such as:
Found /extlinux/extlinux.conf
Retrieving file: /extlinux/extlinux.conf
144 bytes read in 3 ms (46.9 KiB/s)
1:      Debian-Installer
Retrieving file: /initrd.gz
** Reading file would overwrite reserved memory **

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 include/configs/lx2160a_common.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/configs/lx2160a_common.h b/include/configs/lx2160a_common.h
index 2f3b12a575..dc428c6f49 100644
--- a/include/configs/lx2160a_common.h
+++ b/include/configs/lx2160a_common.h
@@ -240,8 +240,8 @@ int select_i2c_ch_pca9547_sec(unsigned char ch);
 
 #define EXTRA_ENV_SETTINGS			\
 	"hwconfig=fsl_ddr:bank_intlv=auto\0"	\
-	"ramdisk_addr=0x9000000\0"		\
-	"ramdisk_addr_r=0x9000000\0"		\
+	"ramdisk_addr=0x85100000\0"		\
+	"ramdisk_addr_r=0x85100000\0"		\
 	"ramdisk_size=0x2000000\0"		\
 	"fdt_high=0xa0000000\0"			\
 	"initrd_high=0xffffffffffffffff\0"	\
-- 
2.34.1

